lanternmq/rabbitmq/messagequeue.go

```diff
@@ -0,0 +1,346 @@
+// over the channel with ID 'chID'. ConsumeFromQueue wraps amqp.Delivery in a lanternmq.Messages object.
+// ConsumeFromQueue creates the receive channel using the RabbitMQ Consume method with the following arguments:
+// queue: qName
+// consumer: ""
+// autoAck: false
+// exclusive: false
+// noLocal: false
+// noWait: false
+// args: nil
+func (mq *MessageQueue) ConsumeFromQueue(chID lanternmq.ChannelID, qName string) (lanternmq.Messages, error) {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return nil, err
+	}
+
+	deliveryChannel, err := ch.Consume(
+		qName, // queue
+		"",    // consumer
+		false, // auto-ack
+		false, // exclusive
+		false, // no-local
+		false, // no-wait
+		nil,   // args
+	)
+	msgs := Messages{deliveryChannel: deliveryChannel}
+
+	return &msgs, err
+}
+
+// ProcessMessages takes 'msgs', which wraps a receive channel for amqp.Delivery objects, and processes each Delivery
+// object by retrieving the message from the Delivery object and providing that along with 'args' to the
+// lanternmq.MessageHandler 'handler'. An acknowledgement is sent to the sender after each message is processed.
+// If there's an error processing a message, the error is sent to the 'errs' channel.
+// ProcessMessages should be called as a gorouting. Example:
+//     go mq.ProcessMessages(msgs, handler, nil, errs)
+func (mq *MessageQueue) ProcessMessages(msgs lanternmq.Messages, handler lanternmq.MessageHandler, args *map[string]interface{}, errs chan<- error) {
+	msgsd, ok := msgs.(*Messages)
+	if !ok {
+		errs <- errors.New("the messages are of the wrong type")
+	}
+
+	for d := range msgsd.deliveryChannel {
+		err := handler(d.Body, args)
+		if err != nil {
+			errs <- err
+		}
+		err = d.Ack(false)
+		if err != nil {
+			errs <- err
+		}
+	}
+}
+
+// DeclareTopic creates a target named 'name' over the channel with ID 'chID'. It uses RabbitMQ's
+// ExchangeDeclare method with the following arguments:
+// name: name
+// kind: "topic"
+// durable: true
+// autoDelete: false
+// intenral: false
+// noWait: false
+// args: nil
+func (mq *MessageQueue) DeclareTopic(chID lanternmq.ChannelID, name string) error {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return err
+	}
+
+	err = ch.ExchangeDeclare(
+		name,    // name
+		"topic", // type
+		true,    // durable
+		false,   // auto-deleted
+		false,   // internal
+		false,   // no-wait
+		nil,     // arguments
+	)
+	if err != nil {
+		err = errors.New("unable to declare target")
+	}
+
+	return err
+}
+
+// PublishToTopic sends 'message' to the topic 'name' over the channel with ID 'chID' with routing key 'routingKey'. It
+// uses RabbitMQ's Publish method with the following arguments:
+// exchange: name
+// key: routingKey
+// mandatory: false
+// immediate: false
+// publishing:
+//   ContentType: "text/plain"
+//   Body: []byte(message)
+func (mq *MessageQueue) PublishToTopic(chID lanternmq.ChannelID, name string, routingKey string, message string) error {
```

Comment:
Use of a config parameter will allow the caller greater flexibility in using the interface/implementation you've defined.